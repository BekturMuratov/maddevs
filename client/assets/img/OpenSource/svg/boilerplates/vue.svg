<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>