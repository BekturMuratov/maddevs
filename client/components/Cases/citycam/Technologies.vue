<template>
  <section class="container_regular">
    <h2 class="case_title_h2 m-24_bottom media-m-12_bottom">
      Technology stack
    </h2>
    <ListTechnologies class="m-96_bottom media-m-48_bottom">
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        class="case_tech-list-item"
        v-bind="technology"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import ListTechnologies from '@/components/Cases/shared/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { technologies } from '@/data/caseCityCam'

export default {
  name: 'Technologies',
  components: {
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>

<style lang="scss" scoped>
.case {
  &_tech-list-item::v-deep {
    p {
      font-size: 12px;
    }

    .django {
      width: 57px;
    }

    @media screen and (max-width: 768px) {
      p {
        font-size: 8px;
      }
      .tech{
        &-texas-instruments,
        &-altium-designer{
          white-space: normal;
          line-height: 7.5px;
        }
      }

      .django {
        width: 30px;
      }
    }
  }
}
</style>
