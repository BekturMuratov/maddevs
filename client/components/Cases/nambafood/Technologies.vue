<template>
  <section class="container_regular">
    <TextParagraph class="m-24_top media-m-48_top">
      Monitoring also helps Mad Devs to prevent service timeouts. When the load grows, we can track spikes, estimate the
      severity of an issue and quickly resolve it.
    </TextParagraph>
    <h3 class="case_title_h3 m-96_top m-48_bottom media-m-72_top media-m-24_bottom">
      Technologies used for the Namba Food project
    </h3>
    <ListTechnologies>
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        :tech-name="technology.name"
        :class-name="technology.className"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/TextParagraph'
import ListTechnologies from '@/components/Cases/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/ListTechnologiesItem'
import { technologies } from '@/data/caseNambafood'

export default {
  name: 'Technologies',
  components: {
    TextParagraph,
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>

<style></style>