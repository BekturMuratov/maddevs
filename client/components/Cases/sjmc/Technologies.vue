<template>
  <section class="container_regular">
    <h3 class="case_title_h3 m-48_bottom media-m-24_bottom">
      Technology stack
    </h3>
    <ListTechnologies class="m-107_bottom media-m-48_bottom">
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        v-bind="technology"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import ListTechnologies from '@/components/Cases/shared/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { technologies } from '@/data/caseSJMC'

export default {
  name: 'Technologies',
  components: {
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>
