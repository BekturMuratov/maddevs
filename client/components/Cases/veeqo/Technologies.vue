<template>
  <section class="container_regular">
    <h2 class="case_title_h2 case_text-align-center m-24_bottom media-m-12_bottom">
      Technology stack
    </h2>
    <ListTechnologies class="m-96_bottom media-m-48_bottom">
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        v-bind="technology"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import ListTechnologies from '@/components/Cases/shared/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { technologies } from '@/data/caseVeeqo'

export default {
  name: 'Technologies',
  components: {
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>
