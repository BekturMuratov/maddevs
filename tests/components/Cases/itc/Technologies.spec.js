import { render, screen } from '@testing-library/vue'
import Technologies from '@/components/Cases/itc/Technologies'

const stubs = ['ListTechnologies', 'ListTechnologiesItem']

describe('Technologies component', () => {
  it('should render correctly', () => {
    const { container } = render(Technologies, {
      stubs,
    })

    expect(screen.getByText('Technology stack')).not.toBeNull()
    expect(container).toMatchSnapshot()
  })
})
