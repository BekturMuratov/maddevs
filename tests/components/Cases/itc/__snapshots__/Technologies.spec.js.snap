// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technologies component should render correctly 1`] = `
<div>
  <section
    class="container_regular"
  >
    <h2
      class="case_title_h2 m-24_bottom media-m-12_bottom"
    >
      
    Technology stack
  
    </h2>
     
    <listtechnologies-stub
      class="m-96_bottom media-m-48_bottom"
    >
      <listtechnologiesitem-stub
        alt="Angular"
        imgalt="Angular"
        name="angular"
      />
      <listtechnologiesitem-stub
        alt="RxJs"
        imgalt="RxJs"
        name="rxjs"
      />
      <listtechnologiesitem-stub
        alt="Material Design"
        imgalt="Material Design"
        name="material-design"
      />
      <listtechnologiesitem-stub
        alt=".NET Core"
        imgalt=".NET Core"
        name="net-core"
      />
      <listtechnologiesitem-stub
        alt="ASP.NET"
        imgalt="ASP.NET "
        name="asp-net"
      />
      <listtechnologiesitem-stub
        alt="GitLab CI/CD"
        imgalt="GitLab CI/CD"
        name="gitlab"
      />
      <listtechnologiesitem-stub
        alt="Docker"
        imgalt="Docker"
        name="docker"
      />
      <listtechnologiesitem-stub
        alt="AWS"
        imgalt="AWS"
        name="aws-second-logo"
      />
      <listtechnologiesitem-stub
        alt="EKS "
        imgalt="EKS"
        name="eks"
      />
      <listtechnologiesitem-stub
        alt="PostgreSQL"
        imgalt="PostgreSQL"
        name="postgresql"
      />
      <listtechnologiesitem-stub
        alt="Sentry"
        imgalt="Sentry"
        name="sentry-black"
      />
      <listtechnologiesitem-stub
        alt="Grafana"
        imgalt="Grafana"
        name="grafana"
      />
    </listtechnologies-stub>
  </section>
</div>
`;
